torch_xla/csrc/tensor.cpp

```diff
@@ -1769,20 +1802,26 @@ std::shared_ptr<XLATensor::Async> XLATensor::SyncTensorsGraphInternal(
     TensorCollectionBarrier(&coll);
     return nullptr;
   }
-  PostOrderData po_data = RunPostOrder(*tensors, &coll);
+  std::vector<torch::lazy::Value> ir_values;
+  std::vector<torch::lazy::BackendDataPtr> tensor_data_vec;
+  FetchTensorData(tensors, coll.config, coll.indices, ir_values,
+                  tensor_data_vec);
+  PostOrderData po_data = RunPostOrder(ir_values, &coll);
   DebugUtil::SaveTensorsGraphInfo("ScheduleSyncTensorsGraph", *tensors,
-                                  &coll.indices);
+                                  &coll.indices, &ir_values);
```

Comment:
why do we have to do this change through, wouldn't using `tensors` and `coll.indices` still works?

Reply:
It may not work actually, since by this time, the tensor's ir_value is reset. Hence, we have to pass the ir_values along with tensors and indices. Currently tensors are required just to get the device of the tensor.

Reply:
Actually, you are right. Revisiting this code made me realize I can just call the SaveTensorsGraph before the reset. Actually we don't need to change this piece of code. Reverting it back